data/model/repository.py

```diff
@@ -672,3 +675,48 @@ def mark_repository_for_deletion(namespace_name, repository_name, repository_gc_
         callback(namespace_name, repository_name)
 
     return marker.id
+
+
+def get_repository_size_and_cache(repo_id: int):
+    query = (
+        RepositorySize.select(Repository.name, Repository.id, RepositorySize.size_bytes)
+        .join(Repository)
+        .where(Repository.id == repo_id)
+    ).get()
```

Comment:
You will need to wrap this query in a try block and catch the NotExistException from peewee.